node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/container/CGroupV2.java

```diff
@@ -27,12 +28,43 @@ public class CGroupV2 {
 
     private static final Logger logger = Logger.getLogger(CGroupV2.class.getName());
     private static final String MAX = "max";
-    public static final String VESPA_CGEXEC_PATH = "/opt/vespa/bin/vespa-cgexec";
 
-    private final FileSystem fileSystem;
+    private final Path rootCgroupPath;
 
     public CGroupV2(FileSystem fileSystem) {
-        this.fileSystem = fileSystem;
+        this.rootCgroupPath = fileSystem.getPath("/sys/fs/cgroup");
+    }
+
+    /**
+     * Wraps {@code command} to ensure it is executed in the given cgroup.
+     *
+     * <p>WARNING: This method must be called only after vespa-cgexec has been installed.</p>
+     *
+     * @param cgroup  The cgroup to execute the command in, e.g. system.slice/wireguard.scope.
+     * @param command The command to execute in the cgroup.
+     * @see #cgroupOf(ContainerId)
+     */
+    public String[] wrapForExecutionIn(String cgroup, String... command) {
+        UnixPath path = new UnixPath(rootCgroupPath.resolve(cgroup).normalize());
+        if (!path.toString().startsWith(rootCgroupPath + "/"))
+            throw new IllegalArgumentException("Invalid cgroup: " + cgroup);
+        // If more than one parent directory needs to be created, subtree_control needs to be fixed, somehow.
+        path.createDirectory();  // No-op if already exists
+
+        String[] fullCommand = new String[3 + command.length];
+        fullCommand[0] = Defaults.getDefaults().vespaHome() + "/bin/vespa-cgexec";
+        fullCommand[1] = "-g";
+        fullCommand[2] = cgroup;
+        System.arraycopy(command, 0, fullCommand, 3, command.length);
+        return fullCommand;
+    }
+
+    /**
+     * Returns the cgroup directory of the Podman container relative the cgroup file system mount point,
+     * and which appears as the root cgroup within the container.
+     */
+    public String cgroupOf(ContainerId containerId) {
```

Comment:
Add this as a getter to `PartialContainer`?

Reply:
I'm thinking it's better to collect the cgroup hierarchy in this file.

Reply:
Changed to `static`.